### PR TITLE
bpo-32360: Replace  OrderedDict  with  dict in inspect.py module

### PR DESCRIPTION
**Pull request by srinivasreddy**
https://bugs.python.org/issue32360

### CODE REVIEW

File: Lib/inspect.py [hunk 1]
```diff
@@ -2560,7 +2560,7 @@ class BoundArguments:
 
     Has the following public attributes:
 
-    * arguments : OrderedDict
+    * arguments : Dict
```
Comment (merwok) [inline Lib/inspect.py]: Should be: dict

(Docs don’t use English capitalization rules for code references like class names)

**Comment by srinivasreddy** [comment]
@merwok  please take a look.

**Comment by srinivasreddy** [comment]
@rhettinger ping.

File: Lib/inspect.py [hunk 2]
```diff
@@ -2759,8 +2759,8 @@ def __init__(self, parameters=None, *, return_annotation=_empty,
 
                     params[name] = param
             else:
-                params = OrderedDict(((param.name, param)
-                                                for param in parameters))
+                params = dict(((param.name, param)
+                               for param in parameters))
```
Comment (merwok) [inline Lib/inspect.py]: Minor: needless extra set of parens here.
```python
params = dict((param.name, param) for param in parameters)
# or simply:
params = {param.name: param for param in parameters}
```

File: Misc/NEWS.d/next/Documentation/2017-12-24-03-29-37.bpo-32360.eZe-ID.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Remove references of OrderedDict in the module `inspect`
```
Comment (merwok) [inline Misc/NEWS.d/next/Documentation/2017-12-24-03-29-37.bpo-32360.eZe-ID.rst]: IMO «reference» is vague and «import» would be clearer here.

File: Lib/inspect.py [hunk 4]
```diff
@@ -49,7 +49,7 @@
 import functools
 import builtins
 from operator import attrgetter
-from collections import namedtuple, OrderedDict
+from collections import namedtuple
```
Comment (merwok) [inline Lib/inspect.py]: If inspect needs to import collections anyway, I don’t think there is much value in removing OrderedDict.
Also replacing `params.move_to_end(p)` by `params[p] = params.pop(p)` without even a comment makes the code a bit worse to read.

I would be -1 on this PR.

(I thought I already left this comment?  Are there duplicate PRs?)

**Comment by srinivasreddy** [comment]
@merwok Done.

**Comment by merwok** [comment]
I did not mean to remove one line in your patch (the import OrderedDict, which in your latest version is an unused import).

I meant that the whole PR seems in the end not needed, since we end up importing collections anyway, and using OrderedDict makes for clearer code.

**Pull request closed without merging**